File: app-android/app/src/main/java/com/health2u/domain/model/Appointment.kt
package com.health2u.domain.model

data class Appointment(
    val id: String,
    val userId: String,
    val title: String,
    val description: String?,
    val doctorName: String?,
    val location: String?,
    val dateTime: Long,
    val reminderMinutes: Int?,
    val status: AppointmentStatus,
    val createdAt: Long
)

enum class AppointmentStatus {
    UPCOMING, COMPLETED, CANCELLED
}

File: app-ios/Sources/Health2u/Domain/Model/Appointment.swift
import Foundation

public enum AppointmentStatus: String, Codable, Sendable, Equatable {
    case upcoming = "UPCOMING"
    case completed = "COMPLETED"
    case cancelled = "CANCELLED"
}

public struct Appointment: Identifiable, Equatable, Sendable {
    public let id: String
    public let userId: String
    public let title: String
    public let description: String?
    public let doctorName: String?
    public let location: String?
    public let dateTime: Date
    public let reminderMinutes: Int?
    public let status: AppointmentStatus
    public let createdAt: Date

    public init(
        id: String,
        userId: String,
        title: String,
        description: String? = nil,
        doctorName: String? = nil,
        location: String? = nil,
        dateTime: Date,
        reminderMinutes: Int? = nil,
        status: AppointmentStatus,
        createdAt: Date
    ) {
        self.id = id
        self.userId = userId
        self.title = title
        self.description = description
        self.doctorName = doctorName
        self.location = location
        self.dateTime = dateTime
        self.reminderMinutes = reminderMinutes
        self.status = status
        self.createdAt = createdAt
    }
}
